File: vol3/vol3-java-examples/src/main/java/com/heatonresearch/aifh/examples/classic/logic/InputNeuron.java
package com.heatonresearch.aifh.examples.classic.logic;

/**
 * A very simple input neuron.
 */
public class InputNeuron implements Neuron {
    /**
     * The value of the input neuron.
     */
    private double value;

    /**
     * @return The current value of the input neuron.
     */
    public double getValue() {
        return value;
    }

    /**
     * Set the value of the input neuron.
     * @param value The new value of the input neuron.
     */
    public void setValue(double value) {
        this.value = value;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public double compute() {
        return this.value;
    }
}

File: vol3/vol3-csharp-examples/AIFH-Vol3/Examples/Classic/Logic/InputNeuron.cs
namespace AIFH_Vol3.Examples.Classic.Logic
{
    /// <summary>
    /// A very simple input neuron.
    /// </summary>
    public class InputNeuron: INeuron
    {
        /// <summary>
        /// The value of the input neuron.
        /// </summary>
        public double Value { get; set; }
        
        /// <inheritdoc/>
        public double Compute()
        {
            return Value;
        }
    }
}
